From c179781cd1bb619167229779e5d9af138ca9b8c4 Mon Sep 17 00:00:00 2001
From: Daniel Leung <daniel.leung@intel.com>
Date: Thu, 24 May 2012 16:54:49 -0700
Subject: Use SSE2 enhanced memset for capable x86 processors

Originally, if TARGET_ARCH_VARIANT is x86-atom, the SSE2 enhanced
memset is used. This patch extends this to all x86 processors which
support SSE2 (i.e. ARCH_X86_HAVE_SSE2 is true).

For: AIA-502
Change-Id: Ibb7ff6ff1ce48da4262e4284618c26f2b516c8bf
Depends-Change-Id: I61c66496df66f6d8adc45cddd48f8750fc95f83c
Signed-off-by: Daniel Leung <daniel.leung@intel.com>
---
 libcutils/Android.mk |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/libcutils/Android.mk b/libcutils/Android.mk
index ceb0d2f..8fee9dd 100644
--- a/libcutils/Android.mk
+++ b/libcutils/Android.mk
@@ -128,12 +128,16 @@ LOCAL_SRC_FILES := $(commonSources) \
 ifeq ($(TARGET_ARCH),arm)
 LOCAL_SRC_FILES += arch-arm/memset32.S
 else  # !arm
-ifeq ($(TARGET_ARCH_VARIANT),x86-atom)
-LOCAL_CFLAGS += -DHAVE_MEMSET16 -DHAVE_MEMSET32
+ifeq ($(TARGET_ARCH),x86)
+ifeq ($(ARCH_X86_HAVE_SSE2),true)
+LOCAL_CFLAGS += -DHAVE_MEMSET16 -DHAVE_MEMSET32 -DUSE_SSE2
 LOCAL_SRC_FILES += arch-x86/android_memset16.S arch-x86/android_memset32.S memory.c
-else # !x86-atom
+else # !ARCH_X86_HAVE_SSE2
 LOCAL_SRC_FILES += memory.c
-endif # !x86-atom
+endif # !ARCH_X86_HAVE_SSE2
+else # !x86
+LOCAL_SRC_FILES += memory.c
+endif # !x86
 endif # !arm
 
 LOCAL_C_INCLUDES := $(libcutils_c_includes) $(KERNEL_HEADERS)
-- 
1.7.4.1

